crates/rust/src/lib.rs

```diff
@@ -884,6 +884,31 @@ impl<'a> wit_bindgen_core::InterfaceGenerator<'a> for InterfaceGenerator<'a> {
 
     fn type_enum(&mut self, id: TypeId, name: &str, enum_: &Enum, docs: &Docs) {
         self.print_typedef_enum(id, name, enum_, docs, &[], Box::new(|_| String::new()));
+
+        let name = name.to_upper_camel_case();
+        let mut cases = String::new();
+        let repr = int_repr(enum_.tag());
+        for (i, case) in enum_.cases.iter().enumerate() {
+            let case = case.name.to_upper_camel_case();
+            cases.push_str(&format!("{i} => {name}::{case},\n"));
+        }
+        uwriteln!(
+            self.src,
+            r#"
+                impl {name} {{
+                    pub(crate) unsafe fn _lift(val: {repr}) -> {name} {{
+                        if !cfg!(debug_assertions) {{
+                            return ::core::mem::transmute(val);
+                        }}
+
+                        match val {{
```

Comment:
this feels like conditionally dead code, I would prefer either an else (which would indent more) or `#[cfg(debug_assertions)]`.

Reply:
True it is conditionally dead code! Personally though I prefer to avoid conditional compilation where possible because that enables rustc to typecheck both implementations and avoids lints such as unused imports as well. The actual codegen-time impact here in theory is quite negligible.